srcpkgs/mercury/template

```diff
@@ -0,0 +1,152 @@
+# Template file for 'mercury'
+pkgname=mercury
+version=22.01
+revision=1
+wrksrc="mercury-srcdist-${version}"
+hostmakedepends="flex bison gettext texinfo"
+depends="mercury-libs>=${version}_${revision}"
+short_desc="Logic/Functional Programming Language"
+maintainer="Emily McDonough <emily@alaskanemily.net>"
+license="GPL-2.0-only"
+homepage="https://mercurylang.org/"
+distfiles="http://dl.mercurylang.org/release/mercury-srcdist-${version}.tar.xz"
+checksum=4ab0734aaa66808c91ffe7f54c7d818bdc4f0247aa255932e1e3d50610db4150
+nocross="Mercury cannot be cross-compiled to different architectures"
+noverifyrdeps=yes
+noshlibprovides=yes
+shlib_requires="libc.so.6"
+
+# asm_fast is preferable, but does not work on all architectures.
+case "$XBPS_MACHINE" in
+	*x86_64*|*arm*)
+		_LLDS_GRADE=asm_fast ;;
+	*i686*)
+		_LLDS_GRADE=reg ;;
+	*)
+		_LLDS_GRADE=none ;;
+esac
+
+_LIBGRADES="hlc.gc,hlc.gc.ssdebug,hlc.gc.memprof,${_LLDS_GRADE}.gc,${_LLDS_GRADE}.gc.stseg,${_LLDS_GRADE}.gc.stseg.decldebug,${_LLDS_GRADE}.gc.stseg.profdeep"
```

Comment:
I would remove `hlc.gc.ssdebug`.

Consider adding `hlc.gc.prof` and `${_LLDS_GRADE}.gc.debug.stseg`.

I suggest listing the grade components in canonical order.

Reply:
> I would remove `hlc.gc.ssdebug`.
I use ssdebug fairly often, it's extremely useful for debugging issues that only happen in HLC, I would prefer to keep it.

> Consider adding `hlc.gc.prof` and `${_LLDS_GRADE}.gc.debug.stseg`.
Isn't decldebug a strict superset of debug?

> I suggest listing the grade components in canonical order.
Is there an actual documented canonical order?

Reply:
Ok, keep ssdebug then.

decldebug is a superset of debug but produces bigger code which is slower to compile. However, a quick check on a project shows that it's only about 6% slower to compile, less than I thought, and the runtime difference is pretty acceptable, too. So it should be fine to skip the .debug grade.

The canonical order is implemented by the canonical_grade script and `mmc --output-grade-string`.

Reply:
OK, I reordered the grades based on its output.